python/rmm/rmm/_lib/device_buffer.pyx

```diff
@@ -138,6 +138,26 @@ cdef class DeviceBuffer:
         }
         return intf
 
+    def prefetch(self, device=None, stream=None):
```

Comment:
*question*: In the C++ interface, the device and stream are required. Do we think it makes sense for them to be required in Python too? Or is the likely usage almost always "fetch on the current device and the stream associated with the buffer"?

Reply:
I was unsure about this. In RMM containers, we make sure the stream is always explicit for stream safety reasons. It's a stream-ordered container so we felt that explicit streams are less error prone.

But in the Python API we have a lot of default values for the stream parameter, so I copied that style here. We may want to revisit this discussion across the whole Python API rather than making a specific decision here that might mismatch current practice?